src/meta/meta_bulk_load_service.cpp

```diff
@@ -822,22 +860,23 @@ void bulk_load_service::update_partition_status_on_remote_storage_reply(
         case bulk_load_status::BLS_INGESTING:
         case bulk_load_status::BLS_SUCCEED:
         case bulk_load_status::BLS_PAUSED:
-            if (old_status != new_status && --_apps_in_progress_count[pid.get_app_id()] == 0) {
+            if (old_status != new_status && !_apps_rolling_back[pid.get_app_id()] &&
+                --_apps_in_progress_count[pid.get_app_id()] == 0) {
                 update_app_status_on_remote_storage_unlocked(pid.get_app_id(), new_status);
             }
             break;
-        case bulk_load_status::BLS_DOWNLOADING:
-            if (old_status != new_status) {
-                _partitions_bulk_load_state.erase(pid);
-                _partitions_total_download_progress[pid] = 0;
-                _partitions_cleaned_up[pid] = false;
-                if (--_apps_in_progress_count[pid.get_app_id()] == 0) {
-                    _apps_in_progress_count[pid.get_app_id()] =
-                        _app_bulk_load_info[pid.get_app_id()].partition_count;
-                    ddebug_f("app({}) restart to bulk load", app_name);
-                }
+        case bulk_load_status::BLS_DOWNLOADING: {
+            _partitions_bulk_load_state.erase(pid);
```

Comment:
Could you explain why did you remove the judgement of `if (old_status != new_status) {` ?

Reply:
In previous implementation, when partition is downloading, it needn't to be updated into downloading again. However, it will raise bug because the downloading progress is not cleaned.